code/modules/vending/vending.dm

```diff
@@ -1586,7 +1616,335 @@
+		//Finds items that have been labeled
+		var/regex/labelFinder = new("\\*? \\(.*?\\)")
+		//Extracts label contents via regex replace
+		var/regex/labelExtractor = new("(?:.*?\\()(.*?)\\)")
+		var/label = null
+		if((target.real_name != target.name) && labelFinder.Find(target.name))
+			label = labelExtractor.Replace(target.name, "$1")
+		//Add the item to an existing entry if there is one
+		for (var/datum/data/vending_product/player_product/R in src.player_list)
+			if(label && label == R.label)
+				R.contents += target
+				R.product_amount += 1
+				existed = 1
+				break
+			else if(istype(target,R.product_type) && R.real_name == target.real_name)
+				R.contents += target
+				R.product_amount += 1
+				existed = 1
+				break
+		if(!existed)
+			var/datum/data/vending_product/player_product/itemEntry = new/datum/data/vending_product/player_product(target, 15)
+			itemEntry.icon = getScaledIcon(target)
+			player_list += itemEntry
+			if(label) itemEntry.label = label
+			generate_slogans()
+
+	generate_wire_HTML()
+		. = ..()
+		var/list/html_parts = list()
+		html_parts += "<table border=\"1\" style=\"width:100%\"><tbody><tr><td><small>"
+		html_parts += "Registered Owner: "
+		if(!owner)
+			html_parts += "<a href='?src=\ref[src];unlock=true'>Unregistered (locked)</a></br>"
+		else
+			html_parts += "<a href='?src=\ref[src];unlock=true'>[src.cardname] "
+			if (!unlocked == 1) html_parts += "(locked) </a></br>"
+			else html_parts += "(unlocked) </a></br>"
+		html_parts += "Loading Chute:  "
+		if(!loading == 0)
+			html_parts += "<a href='?src=\ref[src];loading=false'>Open</a></br> "
+		else
+			html_parts += "<a href='?src=\ref[src];loading=true'>Closed</a></br> "
+		html_parts += "Vendor Name:  "
+		html_parts += "<a href='?src=\ref[src];rename=true'>[src.name]</a> "
+		html_parts += "</small></td></tr></tbody></table></TT><br>"
+		src.wire_HTML += jointext(html_parts, "")
+
+	attackby(obj/item/target, mob/user)
+		if(loading && panel_open)
+			addProduct(target, user)
+		else
+			. = ..()
+		if(!panel_open) //lock up if the service panel is closed
+			loading = 0
+			unlocked = 0
+		src.generate_HTML(1)
+	//Save and restore icon state in case we're in product display mode
+	power_change()
+		var/original_icon = src.icon_state
+		. = ..()
+		if (!(status & BROKEN) && powered())
+			src.icon_state = original_icon
+			setCrtOverlayStatus(1)
+		else
+			setCrtOverlayStatus(0)
+
+	Topic(href, href_list)
+		. = ..()
+		if (status & (BROKEN|NOPOWER))
+			setCrtOverlayStatus(0)
+			return
+		if (usr.stat || usr.restrained() || !in_interact_range(src, usr))
+			return
 
+		if (href_list["loading"])
+			if (src.panel_open && src.unlocked)
+				loading = !loading
+				src.generate_HTML(0, 1)
+		else if (href_list["unlock"] && src.panel_open)
+			if (!owner && src.scan?.registered)
+				owneraccount = FindBankAccountByName(src.scan.registered)
+				owner = src.scan.registered
+				cardname = src.scan.name
+				unlocked = 1
+				loading = 1
```

Comment:
Could it just register to whoever swipes first? Should you be able to... re-register. 😈

Reply:
Unregistering is impossible so the cops can catch you

Reply:
~Emag to clear registered owner?~

Reply:
It randomizez the owner